pkg/kubernetes/standalone/src/cockpit-kube-auth/main.go

```diff
@@ -45,18 +45,18 @@ func jsonError(err error) ([]byte, error) {
 	return json.Marshal(data)
 }
 
-func readData(file *os.File) ([]byte, error) {
+func readData(fd int) ([]byte, error) {
 	result := make([]byte, MAX_BUFFER)
-	n, err := file.Read(result)
-	if err != nil && err != io.EOF {
+	n, err := syscall.Read(fd, result)
```

Comment:
Makes sense. But that's exactly the sort of thing that needs to go in a comment.